app/hooks/useContracts.ts

```diff
@@ -72,6 +72,24 @@ export function useContracts({
 
   const { walletsState, balanceList } = useAppState();
 
+  // Early return with default values if not an EVM chain or missing required props
+  if (!assetId) {
+    return {
+      tokenName: undefined,
+      tokenSymbol: undefined,
+      decimals: undefined,
+      balance: 0,
+      error: undefined,
+      loadMetadata: () => Promise.resolve(),
+      approveSpending: () => Promise.resolve(""),
+      getAllowance: () => Promise.resolve(BigInt(0)),
+      deposit: () => Promise.resolve(""),
+      depositWithExpiry: () => Promise.resolve(""),
+      formatAmount: () => "0",
+      parseAmount: () => BigInt(0),
+    };
+  }
```

Comment:
I’m not entirely sure why we need to initialize the entire object here. Ideally, the hook should only be used when the application is in the correct state. Initializing it this way might just mask issues like unnecessary re-renders or state management problems. If there's an undefined access or something similar happening, I think it would be better to trace the root cause and address it at the state management level rather than adding a default value that could potentially hide the underlying issue. What do you think?